proposals/0327-actor-initializers.md

```diff
@@ -172,27 +182,17 @@ class Process {
   @PIDActor var pid: ProcessID = genPID()
   
   init() {} // Problem: what is the isolation of this init?
-  
-  init() async {} // Problem: no `await` is written to acknowledge
-                  // that to initialize `status` and `pid`, an
-                  // async call would be required.
-
-  deinit {
-    // Problem: how do we release the resources contained
-    // in our global-actor isolated stored properties from 
-    // a deinit, which can never be actor-isolated?
-  }
 }
 ```
 
-In the example above, because `status` and `pid` are isolated to two different global-actors, there's no single actor-isolation that can be specified for the synchronous `init`.
-In fact, all non-delegating initializers would need to have the same isolation as all stored properties.
-For the asynchronous `init`, the fact that a suspension may occur is not explicit in the program, because no `await` is needed on the right-hand side expression of the property declaration's assignment.
-Finally, even if the isolation of the initializers and stored properties matched, the deinit still can _never_ access the stored properties in order to invoke clean-ups routines, without using unsafe lifetime extensions of the actor from the `deinit`.
+In the example above, because `status` and `pid` are isolated to two different global-actors, there's no single actor-isolation that can be specified for the non-async `init`.
+This means that `getStatus` and `genPID` are being called without acquiring actor isolation!
+<!-- In fact, all non-delegating initializers would need to have the same isolation as all stored properties. -->
+<!-- Even if the isolation of the initializers and stored properties matched, the `deinit` still can _never_ access the stored properties in order to invoke clean-up routines, *etc*, without using illegal lifetime extensions of the actor instance from the `deinit`. -->
```

Comment:
Might be worth uncommenting?

Reply:
The point about non-delegating inits matching in isolation was an interesting point for an older example. The current example has two distinct actor isolations, so it's not possible to define even one valid non-async init if those default values are there.

For the deinits, it's just a corollary of the fact that deinits are nonisolated, so I dropped it because it's minor in comparison to something being totally invalid.